src/Symfony/Component/Messenger/Command/FailedMessagesShowCommand.php

```diff
@@ -98,12 +102,18 @@ private function listMessages(SymfonyStyle $io, int $max)
                 $errorMessage = $lastRedeliveryStampWithException->getExceptionMessage();
             }
 
-            $rows[] = [
-                $this->getMessageId($envelope),
-                \get_class($envelope->getMessage()),
-                null === $lastRedeliveryStamp ? '' : $lastRedeliveryStamp->getRedeliveredAt()->format('Y-m-d H:i:s'),
-                $errorMessage,
-            ];
+            $currentClassName = str_replace('\\', '', \get_class($envelope->getMessage()));
+
+            if (!$classFilter || $classFilter === $currentClassName) {
```

Comment:
This check could have been done early: just after `foreach ($envelopes as $envelope) {`